src/main/kotlin/com/fasterxml/jackson/module/kotlin/ConstructorInstantiator.kt

```diff
@@ -0,0 +1,61 @@
+package com.fasterxml.jackson.module.kotlin
+
+import com.fasterxml.jackson.databind.DeserializationContext
+import com.fasterxml.jackson.databind.MapperFeature
+import com.fasterxml.jackson.module.kotlin.Instantiator.Companion.INT_PRIMITIVE_CLASS
+import java.lang.reflect.Constructor
+import kotlin.reflect.KFunction
+import kotlin.reflect.KParameter
+
+// This class does not support inner constructor.
```

Comment:
Does "inner constructor" here mean non-static inner classes' constructors or something else? (non-static inner class constructors are weird and not supporting them makes sense, only asking for clarity)

Reply:
Exactly.
The constructor of the `inner class` requires an instance of the `outer class`, so I was not going to support it in this class.